File: SPRING/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/SupplierService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.dto.request.SupplierRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.response.SupplierResponse;

import java.util.List;

public interface SupplierService {
    SupplierResponse create(SupplierRequest request);
    SupplierResponse update(Long id, SupplierRequest request);
    SupplierResponse getById(Long id);
    List<SupplierResponse> getAll();
    List<SupplierResponse> getActive();
    List<SupplierResponse> search(String keyword);
    List<SupplierResponse> getWithDue();
    void delete(Long id);
}

File: SPRING/Hospital_Management_system_angular/src/app/services/supplier.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { SupplierModel } from '../models/supplier.model';

@Injectable({ providedIn: 'root' })
export class SupplierService {
  private api = environment.apiUrl + 'suppliers';
  constructor(private http: HttpClient) {}
  create(data: SupplierModel): Observable<SupplierModel> { return this.http.post<SupplierModel>(this.api, data); }
  update(id: number, data: SupplierModel): Observable<SupplierModel> { return this.http.put<SupplierModel>(`${this.api}/${id}`, data); }
  getById(id: number): Observable<SupplierModel> { return this.http.get<SupplierModel>(`${this.api}/${id}`); }
  getAll(): Observable<SupplierModel[]> { return this.http.get<SupplierModel[]>(this.api); }
  getActive(): Observable<SupplierModel[]> { return this.http.get<SupplierModel[]>(`${this.api}/active`); }
  search(keyword: string): Observable<SupplierModel[]> { return this.http.get<SupplierModel[]>(`${this.api}/search?keyword=${encodeURIComponent(keyword)}`); }
  getWithDue(): Observable<SupplierModel[]> { return this.http.get<SupplierModel[]>(`${this.api}/due`); }
  delete(id: number): Observable<any> { return this.http.delete(`${this.api}/${id}`); }
}
